### PR TITLE
feat: support single-choice and multi-choices in signup page

### PR DESCRIPTION
**Pull request by okatu-loli**
Fix: https://github.com/casdoor/casdoor/issues/2415

### CODE REVIEW

**Comment by hsluoyz** [comment]
@dacongda plz review

File: web/src/auth/SignupPage.js [hunk 1]
```diff
@@ -624,9 +624,43 @@ class SignupPage extends React.Component {
     } else if (signupItem.name === "Agreement") {
       return AgreementModal.renderAgreementFormItem(application, required, tailFormItemLayout, this);
     } else if (signupItem.name.startsWith("Text ")) {
-      return (
-        <div dangerouslySetInnerHTML={{__html: signupItem.label}} />
-      );
+      if (signupItem.type) {
```
Comment (dacongda) [inline web/src/auth/SignupPage.js:627]: signupItem.type should take effect under fields like gender , bio , tag that can used to select, if these field not exist in signup item, you can add

Comment (okatu-loli) [reply web/src/auth/SignupPage.js:627]: ok，I understand it

**Review by dacongda — APPROVED** [review]

File: web/src/auth/SignupPage.js [hunk 2]
```diff
@@ -618,7 +672,8 @@ class SignupPage extends React.Component {
             },
           ]}
         >
-          <Input className="signup-invitation-code-input" placeholder={signupItem.placeholder} disabled={this.state.invitation !== undefined && this.state.invitation !== ""} />
+          <Input className="signup-invitation-code-input" placeholder={signupItem.placeholder}
```
Comment (hsluoyz) [inline web/src/auth/SignupPage.js:675]: Don't do code formatting for existing code

File: web/src/auth/SignupPage.js [hunk 3]
```diff
@@ -633,7 +688,7 @@ class SignupPage extends React.Component {
           <Button type="primary" htmlType="submit" className="signup-button">
             {i18next.t("account:Sign Up")}
           </Button>
-          &nbsp;&nbsp;{i18next.t("signup:Have account?")}&nbsp;
+            &nbsp;&nbsp;{i18next.t("signup:Have account?")}&nbsp;
```
Comment (hsluoyz) [inline web/src/auth/SignupPage.js:691]: Don't do code formatting for existing code

File: web/src/auth/SignupPage.js [hunk 4]
```diff
@@ -403,27 +454,29 @@ class SignupPage extends React.Component {
                 },
               ]}
             >
-              <Input className="signup-email-input" placeholder={signupItem.placeholder} disabled={this.state.invitation !== undefined && this.state.invitation.email !== ""} onChange={e => this.setState({email: e.target.value})} />
+              <Input className="signup-email-input" placeholder={signupItem.placeholder}
```
Comment (hsluoyz) [inline web/src/auth/SignupPage.js:457]: Don't do code formatting for existing code

File: web/src/auth/SignupPage.js [hunk 5]
```diff
@@ -403,27 +454,29 @@ class SignupPage extends React.Component {
...
-              >
-                <SendCodeInput
-                  className="signup-email-code-input"
-                  disabled={!this.state.validEmail}
-                  method={"signup"}
-                  onButtonClickArgs={[this.state.email, "email", Setting.getApplicationName(application)]}
-                  application={application}
-                />
-              </Form.Item>
+                  <Form.Item
```
Comment (hsluoyz) [inline web/src/auth/SignupPage.js:463]: Don't do code formatting for existing code

File: web/src/auth/SignupPage.js [hunk 6]
```diff
@@ -432,7 +485,8 @@ class SignupPage extends React.Component {
       const renderPhoneItem = () => {
         return (
           <React.Fragment>
-            <Form.Item className="signup-phone" label={signupItem.label ? signupItem.label : i18next.t("general:Phone")} required={required}>
+            <Form.Item className="signup-phone"
```
Comment (hsluoyz) [inline web/src/auth/SignupPage.js:488]: Don't do code formatting for existing code

File: web/src/auth/SignupPage.js [hunk 7]
```diff
@@ -487,26 +541,26 @@ class SignupPage extends React.Component {
...
-                <SendCodeInput
-                  className="signup-phone-code-input"
-                  disabled={!this.state.validPhone}
-                  method={"signup"}
-                  onButtonClickArgs={[this.state.phone, "phone", Setting.getApplicationName(application)]}
-                  application={application}
-                  countryCode={this.form.current?.getFieldValue("countryCode")}
-                />
-              </Form.Item>
+                  <Form.Item
```
Comment (hsluoyz) [inline web/src/auth/SignupPage.js:544]: Don't do code formatting for existing code

File: web/src/auth/SignupPage.js [hunk 8]
```diff
@@ -524,7 +578,7 @@ class SignupPage extends React.Component {
 
         return (
           <React.Fragment>
-            <Row style={{marginTop: "30px", marginBottom: "20px"}} >
+            <Row style={{marginTop: "30px", marginBottom: "20px"}}>
```
Comment (hsluoyz) [inline web/src/auth/SignupPage.js:581]: Don't do code formatting for existing code

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**